pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -779,3 +830,75 @@ func (c *Cluster) applyUpgradeOSDFunctionality() {
 		}
 	}
 }
+
+func (c *Cluster) getOSDStoreUpdateInfo() (*OSDReplaceConfig, *OSDReplaceInfo, error) {
+	if c.spec.Storage.Store.UpdateStore != OSDStoreUpdateConfirmation {
+		return nil, nil, nil
+	}
+
+	replaceConfig, err := c.getOSDReplaceConfig()
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to get OSD replace config")
+	}
+
+	if len(replaceConfig.OSDs) == 0 {
+		logger.Infof("all osds have already been migrated to backend store %q", c.spec.Storage.Store.Type)
+		return nil, nil, nil
+	}
+
+	logger.Infof("%d osd(s) require migration to new backend store %q.", len(replaceConfig.OSDs), c.spec.Storage.Store.Type)
+
+	// replace OSDs only if Pgs are active + clean
+	pgHealthMsg, pgClean, err := cephclient.IsClusterClean(c.context, c.clusterInfo)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to check if the pgs are clean before replacing OSDs")
+	}
+
+	replaceConfig.IsPgHealthy = pgClean
+	if !replaceConfig.IsPgHealthy {
```

Comment:
Why store this in the `replaceConfig`? How about just checking the bool directly?
```suggestion
	if !pgClean {
```